test/metabase/test/data/interface.clj

```diff
@@ -213,21 +213,33 @@
   Currently, this only affects `db-qualified-table-name`."
   nil)
 
+(defn- normalize-qualified-name [n]
+  (-> n u/lower-case-en (str/replace #"-" "_")))
+
+(defn- db-qualified-table-name-prefix [db-name]
+  (str (normalize-qualified-name (or *database-name-override* db-name))
+       \_))
```

Comment:
I extracted this logic out so it could be reused in both `db-qualified-table-name` and `qualified-by-db-name?`, making the coupling explicit